ion-schema/src/main/kotlin/com/amazon/ionschema/model/AstVariablyOccurringType.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model
+
+/**
+ * Represents a type that has an `occurs` value. Note that this includes even
+ * implicit `occurs`, such as in this example:
+ * ```
+ * type::{
+ *   fields: {
+ *     foo: string  // this `string` has an implicit `occurs: optional`
```

Comment:
I am not sure if I follow how this implicit `occurs` is added but for `fields` the default is `occurs: optional` and for `ordered_elements` it is `occurs: required`.

Reply:
In ISL, the `occurs` can be implicit. In this model it's always explicit. It's `val occurs: DiscreteRange` in `AstVariablyOccuringType`.